From d47967f38a8e9ecc575b8ddd7ded222c8f865945 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 23 May 2019 14:54:46 +0800
Subject: [PATCH] systemd: fix avc denials for systemd_resolved_t

Fix:
type=AVC msg=audit(1558592078.610:218): avc:  denied  { map } for
pid=230 comm="systemd-resolve" path="/etc/passwd" dev="vda" ino=31098
scontext=system_u:system_r:systemd_resolved_t
tcontext=system_u:object_r:etc_t tclass=file permissive=0
type=AVC msg=audit(1559180388.297:87): avc:  denied  { read } for
pid=172 comm="systemd-resolve" name="system_bus_socket" dev="tmpfs"
ino=12876 scontext=system_u:system_r:systemd_resolved_t:s0-s15:c0.c1023
tcontext=system_u:object_r:system_dbusd_var_run_t:s0 tclass=sock_file
permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 58238c1..7ae2e76 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -998,10 +998,13 @@ init_pid_filetrans(systemd_resolved_t, systemd_resolved_var_run_t, dir)
 
 dev_read_sysfs(systemd_resolved_t)
 
+files_map_etc_files(systemd_resolved_t)
+
 kernel_dgram_send(systemd_resolved_t)
 kernel_read_crypto_sysctls(systemd_resolved_t)
 kernel_read_kernel_sysctls(systemd_resolved_t)
 kernel_read_net_sysctls(systemd_resolved_t)
+kernel_read_network_state(systemd_resolved_t)
 
 corenet_tcp_bind_generic_node(systemd_resolved_t)
 corenet_tcp_bind_dns_port(systemd_resolved_t)
@@ -1022,6 +1025,8 @@ seutil_read_file_contexts(systemd_resolved_t)
 systemd_log_parse_environment(systemd_resolved_t)
 systemd_read_networkd_runtime(systemd_resolved_t)
 
+files_list_pids(systemd_resolved_t)
+
 init_write_pid_socket(systemd_resolved_t)
 
 optional_policy(`
-- 
2.7.4

